### PR TITLE
Misc fixes

### PR DESCRIPTION
**Pull request by ghedo**
Some of these fix Travis builds.

### CODE REVIEW

File: crypto/engine/eng_dyn.c [hunk 1]
```diff
@@ -510,8 +510,8 @@ static int dynamic_load(ENGINE *e, dynamic_data_ctx *ctx)
      * would also increase opaqueness.
      */
     fns.static_state = ENGINE_get_static_state();
-    CRYPTO_get_mem_functions(fns.mem_fns.malloc_fn, fns.mem_fns.realloc_fn,
-                             fns.mem_fns.free_fn);
+    CRYPTO_get_mem_functions(&fns.mem_fns.malloc_fn, &fns.mem_fns.realloc_fn,
+                             &fns.mem_fns.free_fn);
```
Comment (levitte) [inline crypto/engine/eng_dyn.c]: Oh, I just fixed that, so this parcular change can be removed (or well, just rebase on top of master)

File: engines/e_dasync.c [hunk 2]
```diff
@@ -674,7 +674,7 @@ static int dasync_cipher_init_key_helper(EVP_CIPHER_CTX *ctx,
         pipe_ctx->inner_cipher_data = OPENSSL_zalloc(
             EVP_CIPHER_impl_ctx_size(cipher));
         if (pipe_ctx->inner_cipher_data == NULL) {
-            DASYNCerr(DASYNC_F_DASYNC_AES128_INIT_KEY,
+            DASYNCerr(DASYNC_F_DASYNC_CIPHER_INIT_KEY_HELPER,
```
Comment (richsalz) [inline engines/e_dasync.c]: also fixed in master.

Comment (ghedo) [inline engines/e_dasync.c]: Hmmm, is it? Doesn't look like it's been pushed.

**Comment by mattcaswell** [comment]
+1

**Comment by levitte** [comment]
+1

File: engines/e_dasync.c (hunk 2 above)
Comment (richsalz) [inline engines/e_dasync.c]: Okay, internal pending change, never mind they’ll rebase.

**Comment by mattcaswell** [comment]
Merged. Thanks.

**Pull request closed without merging**